nbody_gravitational/NBodyGravitational.jl

```diff
@@ -0,0 +1,173 @@
+module NBodyGravitational
+
+using DiffEqBase, DifferentialEquations, StaticArrays, Plots, StaticArrays
+
+
+grav_const = 6.673e-11
+
+#=
+    Represents a body/particle in an N-body gravitational problem
+    m - mass
+    r - initial position
+    v - initial velocity
+=#
+struct MassBody{mType, cType}
+    m :: mType
+    r :: Vector{cType}
+    v :: Vector{cType}
+    
+    function MassBody{mType,cType}(m::mType, r::Vector{cType}, v::Vector{cType})
+        @assert length(r)==length(v) "The length of the position (r) and velocity (v) components should match"
+        new(m,r,v)
+    end
+end
+
+MassBody(m::mType, r :: Vector{cType}, v :: Vector{cType}) where {mType <: AbstractFloat, cType  <: AbstractFloat} = MassBody{mType, cType}(m,r,v)
+
+#=
+    Contains the necessary information for solving an N-body gravitational problem
+    bodies - interacting bodies/particles
+    tSpan - time, for which we want to evaluate the solution
+    G - gravitational constant
+=#
+struct NBodyGravProblem
+    bodies :: Array{MassBody, 1}
+    tSpan :: Tuple{Float64, Float64}
+    G :: Float64
```

Comment:
Is there a reason this is localized? While normally globals are bad, here a global constant makes sense.

Reply:
I wanted to create `NBodyGravProblem` as an object representing the task: with its own defined bodies, constants and units of measurements. That is why I let a user define this gravitational constant via constructor of `NBodyGravProblem`.